### DISCUSSION TITLE
some problems of bisection function

### DISCUSSION THREAD

**Discussion by heijiabai** [yalmip/YALMIP#1356] [General discussion]
At present, I have encountered the following problems when using bisection to solve generalized eigenvalues:

 

This is the code:

%%%%%%%%%%%%%%%%%%%

clear;

A =[-1 2; -3 -4];

P = sdpvar(2);

t = sdpvar(1) ;

Constraints = [P>= eye(2), A'*P + P*A <=-2*t*P];

obj = t;

h_opt = bisection(Constraints,obj,sdpsettings('solver','mosek'))

 

%%%%%%%%%%%%%%%%%%%%

I am sorry to found that the result is :

Selected solver: MOSEK-SDP

Generating initial bound: 0 (ok), -1 (ok), -3 (ok), -7 (ok), -15 (ok), -31 (ok), -63 (ok), -127 (ok), -255 (ok), -511 (ok), -1023 (ok), -2047 (ok), -4095 (ok), -8191 (ok), -16383 (ok), -32767 (ok), -65535 (ok), -131071 (ok), -262143 (ok), -524287 (ok), -1048575 (oj). Giving up, always feasible!

 

h_opt = 

 

  包含以下字段的 struct:

 

    yalmiptime: 0.2011

    solvertime: 0.0356

          info: 'Unbounded objective function (learn to debug) (bisection)'

       problem: 2

 

yet the real result is :h_opt = 2.5000

 

I can not  find the cause of the problem, I hope that someone can give me good advices for the problem.Thank you.

**Comment by johanlofberg** [comment]
You want to maximize t, not minimize (which trivially is unbounded)

**Comment by heijiabai** [comment]
Thank you, I will have a try.











黑加白
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yalmip/YALMIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月2日(星期一) 下午3:18
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yalmip/YALMIP] some problems of bisection function (Issue #1334)





 
You want to maximize t, not minimize (which trivially is unbounded)

**Comment by heijiabai** [comment]
I am sorry to found that   https://groups.google.com/g/yalmip  is closed, and I can not understand the explaination for the bisection function in https://yalmip.github.io/command/bisection/

**Comment by johanlofberg** [comment]
It is not closed. 

What is the question though, I already told you that the relevant thing to optimize is to maximize the decay rate, i.e. minimize the negated objective

**Comment by heijiabai** [comment]
Thank you very much.  
The result is from the article, as shown below:
![image](https://github.com/yalmip/YALMIP/assets/146597716/461d9bdd-9f51-40c7-ba9d-d0073190acce)
and from my network, I can not open https://groups.google.com/g/yalmip, as show below:
![image](https://github.com/yalmip/YALMIP/assets/146597716/4c291ab1-65da-44bd-abda-bf1e7f3328e5)

**Comment by johanlofberg** [comment]
It makes no sense to minimize t, as the optimal solution, if system is stable, is -infinity. There is a typo in the text.

Google groups is blocked in China I guess then

**Comment by johanlofberg** [comment]
The original example is here
https://yalmip.github.io/example/decayrate/

**Comment by heijiabai** [comment]
Thank you very very much. OK ,I think I have understand.
